### PR TITLE
[build-properties][docs] update JSDoc comments, update doc page

### PR DESCRIPTION
**Pull request by Simek**
# Why

Update package JSDoc comments to be more meaningful, and more inline our writing guide.

# How

Rewrite and reformat inline comments in source, add platform labels, update documentation page, add an additional example.

# Test Plan

The docs and docs data changes have been tested by running docs app locally.

Update package has been tested by temporary adding it to one of the apps and attempting to run `prebuild`.

# Preview

Screenshot 2023-01-27 at 08 36 33

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by gabrieldonadel — APPROVED** [review]
LGTM

File: docs/pages/versions/unversioned/sdk/build-properties.mdx [hunk 1]
```diff
@@ -9,15 +9,19 @@ import { ConfigPluginExample } from '~/components/plugins/ConfigSection';
 import { APIInstallSection } from '~/components/plugins/InstallSection';
 import PlatformsSection from '~/components/plugins/PlatformsSection';
 
-**`expo-build-properties`** is a [config plugin](/guides/config-plugins) configuring the native build properties of your `ios/Podfile.properties.json` and `android/gradle.properties` directories during [Expo Prebuild](/workflow/prebuild). This config plugin configures how [Expo Prebuild](/workflow/prebuild) generates the native `ios` and `android` folders and therefore cannot be used with projects that don't run `npx expo prebuild` (bare projects).
+**`expo-build-properties`** is a [config plugin](/guides/config-plugins) configuring the native build properties
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/build-properties.mdx:12]:
```suggestion
`expo-build-properties` is a [config plugin](/guides/config-plugins) configuring the native build properties
```

The ** outside of backticks don't render. I think its safe to remove them.

File: docs/pages/versions/unversioned/sdk/build-properties.mdx [hunk 2]
```diff
@@ -45,9 +49,29 @@ import PlatformsSection from '~/components/plugins/PlatformsSection';
...
+      "deploymentTarget": "13.0"
+    }
+  }]
+]);
+```
+
+### All configurable properties
 
-Learn more from [`PluginConfigType`](#pluginconfigtype).
+[`PluginConfigType`](#pluginconfigtype) interface represent all the currently available configuration properties.
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/build-properties.mdx:74]:
```suggestion
[`PluginConfigType`](#pluginconfigtype) interface represents currently available configuration properties.
```

**Review by amandeepmittal — APPROVED** [review]
Thanks for making these changes 👍

File: docs/pages/versions/unversioned/sdk/build-properties.mdx (hunk 1 above)
Comment (Simek) [reply docs/pages/versions/unversioned/sdk/build-properties.mdx:12]: Agree, didn't spot that, nice catch! 👍

**Pull request merged**